crates/lang/ir/src/ir/attrs.rs

```diff
@@ -809,6 +896,23 @@ impl TryFrom<syn::NestedMeta> for AttributeFrag {
                 match &meta {
                     syn::Meta::NameValue(name_value) => {
                         if name_value.path.is_ident("selector") {
+                            if let syn::Lit::Str(lit_str) = &name_value.lit {
+                                let argument = lit_str.value();
+                                // We've pre-processed the verbatim `_` to `"_"`. This was done
+                                // because `syn::Attribute::parse_meta` does not support verbatim.
+                                if argument != "_" {
+                                    return Err(format_err!(
+                                        name_value,
+                                        "#[ink(selector = ..)] attributes with string inputs are deprecated. \
+                                        use an integer instead, e.g. #[ink(selector = 1)] or #[ink(selector = 0xC0DECAFE)]."
+                                    ))
+                                }
+                                return Ok(AttributeFrag {
+                                    ast: meta,
+                                    arg: AttributeArg::Selector(SelectorOrWildcard::Wildcard),
+                                })
+                            }
```

Comment:
I'd appreciate if we create a follow-up issue that fixes this work around with a proper implementation that customizes the argument parsing.